selfdrive/test/process_replay/regen.py

```diff
@@ -117,48 +117,100 @@ def replay_service(s, msgs):
       rk.keep_time()
 
 
-def replay_cameras(lr, frs, disable_tqdm=False):
+def replay_single_camera(s, stream, dt, vipc_server, frames, size):
+  services = [(s, stream)]
+  pm = messaging.PubMaster([s for s, _ in services])
+  rk = Ratekeeper(1 / dt, print_delay_threshold=None)
+
+  img = b"\x00" * int(size[0] * size[1] * 3 / 2)
+  while True:
+    if frames is not None:
+      img = frames[rk.frame % len(frames)]
+
+    rk.keep_time()
+
+    m = messaging.new_message(s)
+    msg = getattr(m, s)
+    msg.frameId = rk.frame
+    msg.timestampSof = m.logMonoTime
+    msg.timestampEof = m.logMonoTime
+    pm.send(s, m)
+
+    vipc_server.send(stream, img, msg.frameId, msg.timestampSof, msg.timestampEof)
+
+  
+def replay_dual_camera(keys, streams, frames, sizes, dt, vipc_server):
+  if len(keys) != len(streams) != len(frames) != len(sizes) != 2:
+    raise ValueError("keys, streams, frames, and sizes must all be length 2")
+
+  pm = messaging.PubMaster(keys)
+  rk = Ratekeeper(1 / dt, print_delay_threshold=None)
+
+  size1, size2 = sizes
+  frames1, frames2 = frames
+
+  img1 = b"\x00" * int(size1[0] * size1[1] * 3 / 2)
+  img2 = b"\x00" * int(size2[0] * size2[1] * 3 / 2)
+  while True:
+    if frames1 is not None:
+      img1 = frames1[rk.frame % len(frames1)]
+    if frames2 is not None:
+      img2 = frames2[rk.frame % len(frames2)]
+    
+    rk.keep_time()
+
+    logMonoTime = int(sec_since_boot() * 1e9)
+    for index in range(len(keys)):
+      s, stream, img = keys[index], streams[index], [img1, img2][index]
+      m = messaging.new_message(s)
+      m.logMonoTime = logMonoTime
+      msg = getattr(m, s)
+      msg.frameId = rk.frame
+      msg.timestampSof = m.logMonoTime
+      msg.timestampEof = m.logMonoTime
+      pm.send(s, m)
+
+      vipc_server.send(stream, img, msg.frameId, msg.timestampSof, msg.timestampEof)
+
+
+def replay_eon_cameras(frs, disable_tqdm=False):
```

Comment:
same here

Reply:
Since the process of simulating camerad, differs so much on eon vs tici (with the new dual cam replay), I thought it would be cleaner to make separate methods for each.